Add support for GenomeSets (as requested for the GSP)

landml opened the pull request:
@JamesJeffryes  - 
Added tests for 'Annotate Multiple Microbial Genomes' and for GenomeSets. Added kb_SetUtilitiesClient.pm for creating a GenomeSet in the tests. If this woks well, the next step will be to add AssemblySets as valid input (PTV-608).

Added GenomeSet as a valid input type.

Deleted two tests that were in the skip directory that are so far out of date that they aren't useful.

landml committed:
Add support for KBaseSearch.GenomeSets

landml committed:
Merge branch 'master' of https://github.com/landml/RAST_SDK

JamesJeffryes commented:
The logic that iterates over the genome set looks good but the genome set will still contain the old genomes. You might want to add in logic that creates a new genome set with the new genomes with "RAST_annotated_" as a prefix. The logic might have to change however as it would be possible to submit multiple genome sets. Might want to check with Dylan about exactly what's needed

landml commented:
Dylan's suggestion was to create a GenomeSet with the output name that the user supplied. Looks like I forgot to add that. Tomorrow.

landml committed:
Add save GenomeSet when annotating multiples

landml committed:
More fixes for creation of output GenomeSet

landml commented:
Added an option to save GenomeSets when annotating multiple genomes with or without an input GenomeSet. Updated the corresponding tests to verify that a new set was created and had the right genome objects. If this works, I'll beef up the testing before adding the AssemblySets.

KBASE-5339 - Made and the display.yaml file parameters consistent. 
@JamesJeffryes - As always, thanks for the helpful comments.

JamesJeffryes commented on ui/narrative/methods/reannotate_microbial_genomes/tmp:
```diff
@@ -0,0 +1,329 @@
+{
```
This file is not needed right?

JamesJeffryes replied:
I'll merge anyway and you can pull this out later

landml replied:
Yup. I can pull that out. Thanks.
